src/ChainSafe.Gaming.Reown/Wallets/ReownWalletRegistry.cs

```diff
@@ -0,0 +1,98 @@
+using ChainSafe.Gaming.Web3.Environment;
+
+namespace ChainSafe.Gaming.Reown.Wallets
+{
+    /// <summary>
+    /// The registry of the wallets supported by Reown.
+    /// </summary>
+    public class ReownWalletRegistry : ILifecycleParticipant, IWalletRegistry // todo provide pre-downloaded registry with option to update
+    {
+        private const string Host = "https://api.web3modal.com";
+
+        private readonly IHttpClient reownHttpClient;
+        private readonly IReownConfig config;
+        private readonly IOperatingSystemMediator systemMediator;
+
+        private List<WalletModel> platformWallets;
+
+        public ReownWalletRegistry(ReownHttpClient reownHttpClient, IReownConfig config, IOperatingSystemMediator systemMediator)
+        {
+            this.systemMediator = systemMediator;
+            this.config = config;
+            this.reownHttpClient = reownHttpClient;
+        }
+
+        public static string RegistryUri => $"{Host}/getWallets";
+
+        public ReadOnlyCollection<WalletModel> SupportedWallets => platformWallets.AsReadOnly();
+
+        async ValueTask ILifecycleParticipant.WillStartAsync()
+        {
+            var registryUri = !string.IsNullOrWhiteSpace(config.OverrideRegistryUri)
+                ? config.OverrideRegistryUri
+                : RegistryUri;
+
+            var parameters = new Dictionary<string, string>()
+            {
+                { "page", 1.ToString() },
+                { "entries", 100.ToString() }, // get as many wallets as we can with one api call (when too many wallets return 400)
+                { "search", null },
+                { "platform", GetPlatformFilter() },
+                { "include", BuildWalletIdsFilter(config.IncludeWalletIds) },
+                { "exclude", BuildWalletIdsFilter(config.ExcludeWalletIds) },
+            };
+
+            var parametersRaw = BuildUriParameters(parameters);
+
+            var response = await reownHttpClient.Get<WalletRegistryResponse>(registryUri + parametersRaw);
+            platformWallets = response.AssertSuccess().Data;
+
+            /* todo implement and utilize IsWalletInstalled functionality
+             *
+                var walletClosure = wallet;
+                var isWalletInstalled = WalletUtils.IsWalletInstalled(wallet);
+             */
+
+            return;
+
+            string GetPlatformFilter()
+            {
+                return systemMediator.Platform switch
+                {
+                    Platform.Android => "android",
+                    Platform.IOS => "ios",
+                    _ => null
+                };
+            }
+
+            string BuildWalletIdsFilter(IList<string> walletIds)
+            {
+                return walletIds is { Count: > 0 }
+                    ? string.Join(",", walletIds)
+                    : null;
+            }
+        }
+
+        private static string BuildUriParameters(Dictionary<string, string> parameters)
```

Comment:
can't we move this to `ReownHttpClient` more in context there imo

Reply:
I would actually even move this to an HttpUtil of some sort

Reply:
Moved it out